starfish/intensity_table.py

```diff
@@ -291,3 +293,123 @@ def mask_small_features(self, size_threshold):
+                set(self.gene_name.values) - {'None'}))
+        int_to_gene[0] = 'None'
+        gene_to_int = {v: k for (k, v) in int_to_gene.items()}
+
+        # map genes to ints
+        gene_list = [gene_to_int[g] for g in self.coords[self.Constants.GENE.value].values]
+        gene_array = np.array(gene_list)
+        decoded_image = gene_array.reshape(self.attrs['image_shape'])  # reverse linearization
+
+        # label the image and extract max intensity across each feature
+        label_image = label(decoded_image, connectivity=2)
+        intensities = self.max(dim=[Indices.CH, Indices.HYB])  # todo can use mean here too
+        intensity_image = intensities.values.reshape(self.attrs['image_shape'])
+
+        # there is a bug in skimage that prevents the use of this method on fake-3d data
+        # see: https://github.com/scikit-image/scikit-image/issues/3278
+        props = regionprops(np.squeeze(label_image), np.squeeze(intensity_image))
+
+        spots = []
+        intensities = []
+        props_passing_filter = []
+
+        # we're dropping spots that fail filters
+        for spot_property in props:
+            if spot_property.area >= area_threshold:
+
+                # because of the above skimage issue, we need to support both 2d and 3d properties
+                if len(spot_property.centroid) == 3:
+                    spot_attrs = {
+                        'z': int(spot_property.centroid[0]),
+                        'y': int(spot_property.centroid[1]),
+                        'x': int(spot_property.centroid[2])
+                    }
+                else:  # data is 2d
+                    spot_attrs = {
+                        'z': 0,
+                        'y': int(spot_property.centroid[0]),
+                        'x': int(spot_property.centroid[1])
+                    }
+
+                # we're back to 3d or fake-3d here
+                gene_index = decoded_image[spot_attrs['z'], spot_attrs['y'], spot_attrs['x']]
+                spot_attrs['gene_name'] = int_to_gene[gene_index]
+                spot_attrs['radius'] = spot_property.equivalent_diameter / 2
+
+                spots.append(spot_attrs)
+                intensities.append(spot_property.max_intensity)
+                props_passing_filter.append(spot_property)
+
+        # now I need to make an IntensityTable from this thing.
+        spots_df = pd.DataFrame(spots)
+
+        # construct coordinate indices
+        spots_index = dataframe_to_multiindex(spots_df)
+        channel_index = [0]
+        round_index = [0]
+
+        # Right now we've eliminated channels and hybs from this tensor, which is not great
+        # but this is therefore a features x 1 x 1 tensor.
+        intensities = np.array(intensities).reshape(-1, 1, 1)
+        dims = (IntensityTable.Constants.FEATURES.value, Indices.CH.value, Indices.HYB.value)
+        attrs = {'image_shape': self.attrs['image_shape']}
+
+        intensity_table = IntensityTable(
```

Comment:
doesn't this need to extend a SpotAttributesTable somewhere?

Reply:
Discussed in slack -- it does not, that will be deprecated upon completion of #305. 👍